app/components/in-viewport/component.js

```diff
@@ -1,41 +1,38 @@
-import Component from "@ember/component";
-import { get } from "@ember/object";
-import classic from "ember-classic-decorator";
-
-@classic
-class InViewportComponent extends Component {
-  rootSelector = "body";
-  rootMargin = 0;
+import { action } from "@ember/object";
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+
+export default class InViewport extends Component {
+  @tracked rootSelector = "body";
+  @tracked rootMargin = 0;
+  @tracked onEnterViewport = this.args["on-enter-viewport"];
+  @tracked onExitViewport = this.args["on-exit-viewport"];
   _observer = null;
 
-  didInsertElement(...args) {
-    super.didInsertElement(...args);
-
+  @action
+  registerObserver(element) {
     const observer = new IntersectionObserver(
       ([{ isIntersecting }]) => {
         if (isIntersecting) {
-          return (get(this, "on-enter-viewport") ?? (() => {}))();
+          return (this.onEnterViewport ?? (() => {}))();
         }
 
-        return (get(this, "on-exit-viewport") ?? (() => {}))();
+        return (this.onExitViewport ?? (() => {}))();
```

Comment:
Access the provided event handlers via `this.args.onEnterViewport`, we don't to store them locally, do we?